src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
-                        Accept(last);
+                        if (!startOfLine || last.Type != HtmlSymbolType.WhiteSpace)
+                        {
+                            Accept(last);
+                        }
+
                         last = null;
                     }
+
                     AddMarkerSymbolIfNecessary();
                     Output(SpanKind.Markup);
+
                     RazorComment();
+
+                    // Handle the whitespace and newline at the end of a razor comment.
+                    if (startOfLine &&
+                        (At(HtmlSymbolType.NewLine) ||
+                        (At(HtmlSymbolType.WhiteSpace) && NextIs(HtmlSymbolType.NewLine))))
+                    {
+                        AcceptWhile(IsSpacingToken(includeNewLines: false));
```

Comment:
This should consume the newline instead of skipping it below (if it's skipped it'll mess up tooling).